modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_22_0/ObjectDefinitionUpgradeProcess.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.internal.upgrade.v3_22_0;
+
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Selton Guedes
+ */
+public class ObjectDefinitionUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (hasColumn("ObjectDefinition", "comments")) {
+			return;
+		}
+
+		alterTableAddColumn("ObjectDefinition", "comments", "BOOLEAN");
+
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
```

Comment:
Hey @seltong,
I was looking if there was a way to improve the code and avoid the TRY statement. I found some upgrades that added a boolean column and did like this:
`			runSQL("update RemoteAppEntry set customElementUseESM = [$FALSE$]");
`
I found some examples: AppUpgradeProcess.java / RemoteAppEntryUpgradeProcess.java / FragmentEntryUpgradeProcess.java.

I think both upgrades could be done in 1 upgrade class and we could try doing in 1 SQL statement, like:
 `			runSQL("update ObjectDefinition set categorization = [$TRUE$], comments = [$FALSE$] ");
`
Is there a reason for you choosing to do in 2 classes?

Please, let me know what you think :smiley:

Reply:
It makes sense! That's great!

Reply:
I saw now that categorization and comments can't be used in SystemObjectDefinitions. So, what do you think about to keep with my approach and add a condition in the update to make sure that categorization and comments will be always false when the ObjectDefinition is system?